internal/zinc-persist/src/main/scala/sbt/internal/inc/FileAnalysisStore.scala

```diff
@@ -50,21 +50,21 @@ object FileAnalysisStore {
      * Get `CompileAnalysis` and `MiniSetup` instances for current `Analysis`.
      */
     override def get: Optional[AnalysisContents] = {
-      import JavaInterfaceUtil.EnrichOption
-      val nestedRead: Option[Option[AnalysisContents]] = allCatch.opt {
+      if (file.exists()) {
         Using.zipInputStream(new FileInputStream(file)) { inputStream =>
           lookupEntry(inputStream, analysisFileName)
           val reader = CodedInputStream.newInstance(inputStream)
           val (analysis, miniSetup) = format.read(reader)
-          val analysisWithAPIs = allCatch.opt {
+          val analysisWithAPIs = {
```

Comment:
Instead of changing the existing semantics, why don't we add another API that returns `Either[Option[String], AnalysisContents]` or something like that, so we can surface the error as a warning message.

Reply:
Could we do better with `Either[AnalysisException, AnalysisContents]` or `Try[AnalysisContents]`? Using Either preserves a bit more type information than Try. I made up AnalysisException. It would be whatever exception type the api throws.

Reply:
I included `Option[String]` in there because analysis file not existing is a case that needs to be handled. Maybe it's better to have `Try[Option[AnalysisContents]]` or `Either[something, Option[AnalysisContents]]`.

Reply:
Oh, I see. I just saw the title was "Do not swallow exception" and assumed there was some exception that was being caught and dropped so I was just advocating preserving that information since `Option[String]` implied a loss of information to me.

Reply:
@eatkins You assumed correctly. `allCatch.opt` does catch and drop exceptions. These are not expected exceptions (like for missing file) but some rare cases like e.g. corrupt analysis file or ClassDefNotFoundError or any other random issues. Currently they will just be ignored and treated equivalently to missing file.
@eed3si9n It would still be an improvement to add separate method (any suggested name), though I believe this new method should be called by zinc internally, because otherwise the error will be lost as before. I would be in favour of throwing the exception, but I guess logging could be ok, yet harder to spot.

Reply:
Once there's `Try` or `Either` the consumer of Zinc can take appropriate action, including failing the build etc..

Reply:
I tried doing that. It seems that adding the new method would require altering the api of `xsbti.compile.AnalysisStore` and I think scala types like `Try` or `Either` shouldn't be used there.
I was thinking about adding `getUnsafe` variant that would throw, and keep `get` as `try getUnsafe() catch { case e: Exception => Optional.empty() }`. Though it feels a bit odd that (in my opinion) the preferred method would be the unsafe variant. I still think that the allCatch there was not the best decision and should just be replaced as it currently is in this PR.